homeassistant/util/package.py

```diff
@@ -40,14 +41,25 @@ def is_installed(requirement_str: str) -> bool:
     expected input is a pip compatible package specifier (requirement string)
     e.g. "package==1.0.0" or "package>=1.0.0,<2.0.0"
 
+    For backward compatibility, it also accepts a URL with a fragment
+    e.g. "git+https://github.com/pypa/pip#pip>=1"
+
     Returns True when the requirement is met.
     Returns False when the package is not installed or doesn't meet req.
     """
     try:
         req = Requirement(requirement_str)
     except InvalidRequirement:
-        _LOGGER.error("Invalid requirement '%s'", requirement_str)
-        return False
+        try:
+            # This was originally used to install zip files, and
+            # we no longer use this in Home Assistant. However, custom
+            # components use it to installed packages from git
+            # urls with a fragment.
+            # example: git+https://github.com/pypa/pip#pip>=1
```

Comment:
```suggestion
            # fragment support was originally used to install zip files, and
            # we no longer do this in Home Assistant. However, custom
            # components started using it to installed packages from git
            # urls with a fragment because it worked so it would be a breaking
            # change to remove it.
            # example: git+https://github.com/pypa/pip#pip>=1
```